default.nix

```diff
@@ -1,16 +1,45 @@
-{ pkgs ? import <nixpkgs> {} }: let
-  overlay = import ./overlay.nix;
+{
+  compiler ? "ghc863"
+, nixpkgs-rev ? "120eab94e0981758a1c928ff81229cd802053158"
+, nixpkgs-sha256 ? "0qk6k8gxx5xlkyg05dljywj5wx5fvrc3dzp4v2h6ab83b7zwg813"
+
+, pkgs ? import (builtins.fetchTarball {
+    url = "https://github.com/NixOS/nixpkgs/archive/${nixpkgs-rev}.tar.gz";
+    sha256 = nixpkgs-sha256;
+  }) {
+    config.allowUnfree = true;
+    config.allowBroken = false;
+  }
+
+, doStrict  ? false
+, doTracing ? false
+
+}:
+
+let
+
+  overlay = self: super: builtins.mapAttrs (k: v: (
```

Comment:
Rather than doing mapAttrs here maybe we should generate both this and the underlying overlay from a list of package names?